crates/rgkl/src/main.rs

```diff
@@ -84,9 +84,8 @@ fn main() -> ExitCode {
     match Signals::new([SIGINT, SIGTERM]) {
         Ok(mut signals) => {
             thread::spawn(move || {
-                for _sig in signals.forever() {
-                    //let _ = term_tx.send(());
-                    break;
+                if signals.forever().next().is_some() {
+                    let _ = term_tx.send(());
```

Comment:
term_tx.send() shouldn't get called because the actual termination action is the drop(term_tx) that comes after